pkg/webhook/controlplane/etcd.go

```diff
@@ -134,3 +144,61 @@ func GetETCDVolumeClaimTemplate(name string, storageClassName *string, storageCa
 		},
 	}
 }
+
+// DetermineBackupSchedule determines the backup schedule based on the shoot creation and maintainance time window.
+func DetermineBackupSchedule(c *corev1.Container, cluster *extensionscontroller.Cluster, defaultSchedule *string) (string, error) {
+	schedule, err := determineExistingBackupSchedule(c)
+	if err != nil {
+		return "", err
+	}
+
+	if len(schedule) != 0 {
+		return schedule, nil
+	}
+
+	// If default schedule is configured, then we use it.
+	if defaultSchedule != nil && len(*defaultSchedule) > 0 {
+		return *defaultSchedule, nil
+	}
+
+	if cluster.Shoot.Spec.Maintenance != nil && cluster.Shoot.Spec.Maintenance.TimeWindow != nil {
```

Comment:
FYI: There is https://github.com/gardener/gardener-extensions/pull/319 which will probably be merged first, and you should then adapt this part to be capable of working with both `cluster.Shoot` and `cluster.CoreShoot`.

Reply:
Cool. Thanks for prior info.